packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -144,29 +144,8 @@ export class WorkspaceMainImpl implements WorkspaceMain {
 
     $startFileSearch(includePattern: string, excludePatternOrDisregardExcludes?: string | false,
         maxResults?: number, token?: theia.CancellationToken): Promise<UriComponents[]> {
-        const uris: UriComponents[] = new Array();
-        let j = 0;
-        // tslint:disable-next-line:no-any
-        const promises: Promise<any>[] = new Array();
-        for (const root of this.roots) {
-            promises[j++] = this.fileSearchService.find(includePattern, { rootUri: root.uri }).then(value => {
-                const paths: string[] = new Array();
-                let i = 0;
-                value.forEach(item => {
-                    let path: string;
-                    path = root.uri.endsWith('/') ? root.uri + item : root.uri + '/' + item;
-                    paths[i++] = path;
-                });
-                return Promise.resolve(paths);
-            });
-        }
-        return Promise.all(promises).then(value => {
-            let i = 0;
-            value.forEach(path => {
-                uris[i++] = Uri.parse(path);
-            });
-            return Promise.resolve(uris);
-        });
+        return this.fileSearchService.find(includePattern, { rootUris: this.roots.map(r => r.uri) })
```

Comment:
maybe use await for readability ?

Reply:
`return await` would trigger an eslint warning. To make it more readable I can do this (this is also what some other projects do)
```ts
return Promise.resolve()
   .then(() => this.fileSearchService.find(....))
   .then(.....);
```
With this pattern, people would tell the function returns a Promise right away.

Reply:
My idea was to use two lines. One for await on fileSearchService, another with return/map
Anyway up to you I approved current code. Was just to avoid long callbacks line